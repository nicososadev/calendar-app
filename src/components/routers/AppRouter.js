import React from 'react'
import { BrowserRouter as Router, Switch, Redirect, Route } from 'react-router-dom'
import { LoginPage } from '../auth/LoginPage'
import { CalendarPage } from '../calendar/CalendarPage'

export const AppRouter = () => {

    return (
        
        <Router>
            <div>
                <Switch>
                    <Route path='/login' component={ LoginPage } />
                    <Route exact path='/' component={ CalendarPage } />

                    <Redirect to='/' />
                </Switch>
            </div>
        </Router>
    )
}
