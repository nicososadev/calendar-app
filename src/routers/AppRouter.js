import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { BrowserRouter as Router, Switch, Redirect } from 'react-router-dom'
import { checkingManager } from '../actions/auth'
import { LoginPage } from '../components/auth/LoginPage'
import { CalendarPage } from '../components/calendar/CalendarPage'
import { PrivateRoute } from './PrivateRoute'
import { PublicRoute } from './PublicRoute'

export const AppRouter = () => {

    const dispatch = useDispatch()

    const { checking, uid } = useSelector(state => state.auth)

    useEffect(() => {
        dispatch( checkingManager() )
        
    }, [dispatch])

    if ( checking ) {

        return (<h4>Wait...</h4>)

    }

    return (
        
        <Router>
            <div>
                <Switch>
                    <PublicRoute exact path='/login' isAuthenticated={ !!uid } component={ LoginPage } />
                    <PrivateRoute exact path='/' isAuthenticated={ !!uid } component={ CalendarPage } />

                    <Redirect to='/' />
                </Switch>
            </div>
        </Router>
    )
}
