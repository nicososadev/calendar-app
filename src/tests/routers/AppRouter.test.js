import '@testing-library/jest-dom'
import { mount } from 'enzyme'
import React from 'react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'
import { AppRouter } from '../../routers/AppRouter'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

describe('Pruebas en el componente AppRouter', () => {
    
    test('Debe mostrarse correctamente ', () => {
        
        const initialState = {
            auth: {
                checking: true
            }
        }
        
        let store = mockStore(initialState)
        
        const wrapper = mount(
            <Provider store={store}>
                <AppRouter />
            </Provider>
        )

        expect(wrapper).toMatchSnapshot()

    })

    test('Debe mostrarse la ruta publica ', () => {
        
        const initialState = {
            auth: {
                checking: false
            }
        }
        
        let store = mockStore(initialState)
        
        const wrapper = mount(
            <Provider store={store}>
                <AppRouter />
            </Provider>
        )

        expect(wrapper).toMatchSnapshot()
        
        const loginContainer = wrapper.find('.login-container')

        expect( loginContainer.exists() ).toBe( true )
        
    })

    test('Debe mostrarse la ruta privada ', () => {
        
        const initialState = {
            ui: {
                modalOpen: false,
            },
            calendar: {
                events: [],
                activeEvent: null
            },
            auth: {
                checking: false,
                uid: '123',
                name: 'test'
            }
            
        }
        
        let store = mockStore(initialState)
        
        const wrapper = mount(
            <Provider store={store}>
                <AppRouter />
            </Provider>
        )

        expect(wrapper).toMatchSnapshot()
        
    })
    
})
