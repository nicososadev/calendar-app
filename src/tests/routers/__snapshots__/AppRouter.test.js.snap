// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AppRouter Debe mostrarse correctamente  1`] = `
<h4>
  Wait...
</h4>
`;

exports[`Pruebas en el componente AppRouter Debe mostrarse la ruta privada  1`] = `
<div>
  <div>
    <div
      className="navbar navbar-dark bg-dark mb-4"
    >
      <span
        className="navbar-brand"
      >
        test
      </span>
      <button
        className="btn btn-danger"
        onClick={[Function]}
      >
        <i
          className="fas fa-sign-out-alt"
        />
        <span>
           Salir
        </span>
      </button>
    </div>
    <div
      className="rbc-calendar"
      style={
        Object {
          "height": 850,
        }
      }
    >
      <div
        className="rbc-toolbar"
      >
        <span
          className="rbc-btn-group"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Today
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Back
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </span>
        <span
          className="rbc-toolbar-label"
        >
          October 2020
        </span>
        <span
          className="rbc-btn-group"
        >
          <button
            className="rbc-active"
            key="month"
            onClick={[Function]}
            type="button"
          >
            Month
          </button>
          <button
            className=""
            key="week"
            onClick={[Function]}
            type="button"
          >
            Week
          </button>
          <button
            className=""
            key="day"
            onClick={[Function]}
            type="button"
          >
            Day
          </button>
          <button
            className=""
            key="agenda"
            onClick={[Function]}
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        className="rbc-month-view"
      >
        <div
          className="rbc-row rbc-month-header"
        >
          <div
            className="rbc-header"
            key="header_0"
          >
            <span>
              Sun
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_1"
          >
            <span>
              Mon
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_2"
          >
            <span>
              Tue
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_3"
          >
            <span>
              Wed
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_4"
          >
            <span>
              Thu
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_5"
          >
            <span>
              Fri
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_6"
          >
            <span>
              Sat
            </span>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  03
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  04
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  05
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  06
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  07
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  08
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  09
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  10
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-today"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  11
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-now rbc-current"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  12
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  13
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  14
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  15
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  16
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  17
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  18
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  19
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  20
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  21
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  22
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  23
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  24
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  25
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  26
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  31
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <button
        className="btn btn-primary fab"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus"
        />
      </button>
    </div>
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        />
      }
    />
  </div>
</div>
`;

exports[`Pruebas en el componente AppRouter Debe mostrarse la ruta publica  1`] = `
<div>
  <div
    className="container login-container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 login-form-1"
      >
        <h3>
          Sign in
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="loginEmail"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="loginPassword"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-6 login-form-2"
      >
        <h3>
          Register
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerName"
              onChange={[Function]}
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerEmail"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerPassword"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerPassword2"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Create account"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
